### PR TITLE
🔥 Discord Connect Fix

### PR DESCRIPTION
**Pull request by Adhikram**

### CODE REVIEW

**Review by vedansh — APPROVED** [review]

File: devsnest/app/workers/mass_notifier_worker.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# worker that appends tokens to a message
+class MassNotifierWorker
+  include Sidekiq::Worker
+
+  def perform(event_message)
+    start = 0
+    while start * 20 <= User.all.count
```
Comment (vedansh) [inline devsnest/app/workers/mass_notifier_worker.rb:9]: 20 is a magical number, remove it, use `NotificationBot.count`

File: devsnest/app/workers/mass_notifier_worker.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# worker that appends tokens to a message
+class MassNotifierWorker
+  include Sidekiq::Worker
+
+  def perform(event_message)
+    start = 0
+    bucket_size = 20
+    while start * NotificationBot.count * bucket_size <= User.count
+      NotificationBot.all.each do |n|
```
Comment (vedansh) [inline devsnest/app/workers/mass_notifier_worker.rb:11]: rename n to bot or a more meaningful name

Comment (Adhikram) [reply devsnest/app/workers/mass_notifier_worker.rb:11]: okay

**Review by kaydee0502 — APPROVED** [review]

**Review by LakshitXD — APPROVED** [review]

**Review by Yug-gurnani — CHANGES_REQUESTED** [review]
Write Rspecs for this

File: devsnest/app/workers/mass_notifier_worker.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def perform(event_message)
+    start = 0
+    bucket_size = 20
+    while start * NotificationBot.count * bucket_size <= User.count
+      NotificationBot.all.each do |bot|
+        discord_ids = User.where(bot_id: bot.id).slice(start * bucket_size, bucket_size)
+        next unless discord_ids.present?
+
+        data = {
+          bot: bot.bot_token, message: event_message,
```
Comment (vedansh) [inline devsnest/app/workers/mass_notifier_worker.rb:16]: every item in new line

File: devsnest/app/workers/mass_notifier_worker.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# worker that appends tokens to a message
+class MassNotifierWorker
+  include Sidekiq::Worker
+
+  def perform(event_message)
+    start = 0
+    bucket_size = 20
+    while start * NotificationBot.count * bucket_size <= User.count
+      NotificationBot.all.each do |bot|
+        discord_ids = User.where(bot_id: bot.id).slice(start * bucket_size, bucket_size)
```
Comment (vedansh) [inline devsnest/app/workers/mass_notifier_worker.rb:12]: `discord_ids = User.where(bot_id: bot.id).slice(start * bucket_size, bucket_size).pluck(:discord_id)`

**Pull request merged**